src/main/java/fi/otavanopisto/kuntaapi/server/integrations/management/ManagementBannerEntityUpdater.java

```diff
@@ -97,90 +112,138 @@ public void stopTimer() {
-    for (Banner managementBanner : managementBanners) {
-      updateManagementBanner(organizationId, api, managementBanner);
-    }
-  }
-
-  private List<Banner> listManagementBanners(DefaultApi api, OrganizationId organizationId) {
-    fi.otavanopisto.mwp.client.ApiResponse<List<Banner>> response = api.wpV2BannerGet(null, null, null, null, null, null, null, null, null, null, null, null, null, null);
+  private void updateManagementBanner(DefaultApi api, OrganizationId organizationId, BannerId managementBannerId) {
+    fi.otavanopisto.mwp.client.ApiResponse<Banner> response = api.wpV2BannerIdGet(managementBannerId.getId(), null);
     if (response.isOk()) {
-      return response.getResponse();
+      updateManagementBanner(api, organizationId, response.getResponse());
     } else {
-      logger.warning(String.format("Listing organization %s banners failed on [%d] %s", organizationId.getId(), response.getStatus(), response.getMessage()));
+      logger.warning(String.format("Finding organization %s banner failed on [%d] %s", managementBannerId.getId(), response.getStatus(), response.getMessage()));
     }
-    
-    return Collections.emptyList();
   }
   
-  private void updateManagementBanner(OrganizationId organizationId, DefaultApi api, Banner managementBanner) {
+  private void updateManagementBanner(DefaultApi api, OrganizationId organizationId, Banner managementBanner) {
     BannerId bannerId = new BannerId(organizationId, ManagementConsts.IDENTIFIER_NAME, String.valueOf(managementBanner.getId()));
 
     Identifier identifier = identifierController.findIdentifierById(bannerId);
     if (identifier == null) {
       identifier = identifierController.createIdentifier(bannerId);
     }
     
-    modificationHashCache.put(identifier.getKuntaApiId(), createPojoHash(managementBanner));
+    BannerId bannerKuntaApiId = new BannerId(organizationId, KuntaApiConsts.IDENTIFIER_NAME, identifier.getKuntaApiId());
+    
+    fi.otavanopisto.kuntaapi.server.rest.model.Banner banner = managementTranslator.translateBanner(bannerKuntaApiId, managementBanner);
+    if (banner == null) {
+      logger.severe(String.format("Could not translate banner %d into Kunta API banner", managementBanner.getId()));
+      return;
+    }
+    
+    bannerCache.put(bannerKuntaApiId, banner);
+    modificationHashCache.put(identifier.getKuntaApiId(), createPojoHash(banner));
     
     if (managementBanner.getFeaturedMedia() != null && managementBanner.getFeaturedMedia() > 0) {
-      updateFeaturedMedia(organizationId, api, managementBanner.getFeaturedMedia()); 
+      updateFeaturedMedia(organizationId, api, bannerKuntaApiId, managementBanner.getFeaturedMedia()); 
     }
+
   }
   
-  private void updateFeaturedMedia(OrganizationId organizationId, DefaultApi api, Integer featuredMedia) {
+  private void updateFeaturedMedia(OrganizationId organizationId, DefaultApi api, BannerId bannerId, Integer featuredMedia) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "updateFeaturedMedia" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)